src/Ledger/Gov.lagda

```diff
@@ -81,6 +82,40 @@ private variable
 \end{code}
 \emph{Functions used in the GOV rules}
 \begin{AgdaMultiCode}
+\begin{code}[hide]
+govActionPriority : GovAction → ℕ
+govActionPriority NoConfidence             = 0
+govActionPriority (UpdateCommittee _ _ _)  = 1
+govActionPriority (NewConstitution _ _)    = 2
+govActionPriority (TriggerHF _)            = 3
+govActionPriority (ChangePParams _)        = 4
+govActionPriority (TreasuryWdrl _)         = 5
+govActionPriority Info                     = 6
+
+groupGovActions : GovState → GovState
+groupGovActions gs  =   filter (λ x → govActionPriority (action (proj₂ x)) ≟ 0) gs
+                    ++  filter (λ x → govActionPriority (action (proj₂ x)) ≟ 1) gs
+                    ++  filter (λ x → govActionPriority (action (proj₂ x)) ≟ 2) gs
+                    ++  filter (λ x → govActionPriority (action (proj₂ x)) ≟ 3) gs
+                    ++  filter (λ x → govActionPriority (action (proj₂ x)) ≟ 4) gs
+                    ++  filter (λ x → govActionPriority (action (proj₂ x)) ≟ 5) gs
+                    ++  filter (λ x → govActionPriority (action (proj₂ x)) ≟ 6) gs
+
+insertGovAction : GovState → GovActionID × GovActionState → GovState
```

Comment:
This also needs to be visible. Also, the `with` isn't that great there, since we'd have to explain it. But I think you should be able to do
```
if govActionPriority (action gaSt₀) ≤ govActionPriority (action gaSt₁) then ... else ...
```
and I expect that this probably doesn't break any proofs.